packages/swap/src/routes/v2/quote.ts

```diff
@@ -225,25 +230,50 @@ export const generateQuotes = async ({
     quoteType: 'REGULAR' as const,
   };
 
-  const [quoteResult, boostedQuoteResult, dcaQuoteResult, dcaBoostedQuoteResult] =
-    await Promise.allSettled([
-      getPoolQuote(quoteArgs),
-      estimatedBoostFeeBps && getPoolQuote({ ...quoteArgs, boostFeeBps: estimatedBoostFeeBps }),
-      dcaQuoteParams &&
-        getPoolQuote({
-          ...quoteArgs,
-          swapInputAmount: dcaQuoteParams.chunkSize,
-          quoteType: 'DCA',
-        }),
-      dcaQuoteParams &&
-        estimatedBoostFeeBps &&
-        getPoolQuote({
-          ...quoteArgs,
-          boostFeeBps: estimatedBoostFeeBps,
-          swapInputAmount: dcaQuoteParams.chunkSize,
-          quoteType: 'DCA',
-        }),
-    ]);
+  const [quoteResult, boostedQuoteResult] = await Promise.allSettled([
+    getPoolQuote(quoteArgs),
+    estimatedBoostFeeBps && getPoolQuote({ ...quoteArgs, boostFeeBps: estimatedBoostFeeBps }),
+  ]);
+  const ingressFee =
+    quoteResult.status === 'fulfilled'
+      ? quoteResult.value.includedFees.find((fee) => fee.type === 'INGRESS')
+      : undefined;
+  const egressFee =
+    quoteResult.status === 'fulfilled'
+      ? quoteResult.value.includedFees.find((fee) => fee.type === 'EGRESS')
+      : undefined;
+
+  const ingressFeeSurcharge = dcaQuoteParams
```

Comment:
```suggestion
  // the swap_rate rpc will deduct the full ingress fee before simulating the swap
  // as we quote a single chunk, we add a surcharge so that the effective deducted amount is 1/numberOfChunks
  const ingressFeeSurcharge = dcaQuoteParams
```